### PR TITLE
Use `yum history rollback` to restore yum state after tests:

### PR DESCRIPTION
**Pull request by benjamreis**
Previous version didn't support downgrade to a previous version, or install disbaled repo (testing, update etc...)
This version rollback to the last yum transaction before the tests and check that the same RPMs with same versions, releases, archs and epochs are installed.

### CODE REVIEW

File: lib/common.py [hunk 1]
```diff
@@ -329,9 +351,10 @@ def yum_remove_added_packages(self, base_packages):
             print("No added packages to remove.")
 
     def yum_restore_saved_state(self):
-        """ Currently only able to remove added packages """
-        assert self.saved_packages_list is not None, "Can't restore previous state without a package list"
-        self.yum_remove_added_packages(self.saved_packages_list)
+        """ Restore yum state to saved state """
+        assert self.saved_packages_list is not None and self.saved_rollback_id, "Can't restore previous state without a package list"
```
Comment (stormi) [inline lib/common.py:355]: I'd rather have two different asserts, one for package list and one for rollback id, each with its own message.

File: lib/common.py [hunk 2]
```diff
@@ -329,9 +351,10 @@ def yum_remove_added_packages(self, base_packages):
...
+        self.ssh(['yum', 'history', 'rollback', '--enablerepo=*', self.saved_rollback_id, '-y'])
+        assert self.saved_packages_list == self.packages()
```
Comment (stormi) [inline lib/common.py:357]: If it fails, the output will be very hard to read. I would prefer a diff and assert that the diff is empty.

Comment (benjamreis) [reply lib/common.py:357]: What about keeping this assert, with an error message ?

Comment (benjamreis) [reply lib/common.py:357]: Diffing would add algo complexity to the tests

Comment (stormi) [reply lib/common.py:358]: I think it's worth it to make the output usable. By the way you don't need to do the detailed comparison before you detect that there's a difference. When it's OK, the complexity will not be higher.

**Pull request merged**